subsys/bluetooth/host/Kconfig

```diff
@@ -585,6 +585,17 @@ config BT_SMP_DISABLE_LEGACY_JW_PASSKEY
 	  This option disables Just Works and Passkey legacy pairing methods to
 	  increase security.
 
+config BT_SMP_ALLOW_UNAUTH_OVERWRITE_AUTHENTICATED
+	bool "Allow unauthenticated pairing for previous authenticated paired device"
```

Comment:
Suggest to make this an experimental Kconfig option, so we can easily remove it again if it turns out to be an issue